### PR TITLE
681 Add  Tensor support for DataStats transform

### LINKED ISSUE #681: Add Tensor support to DataStats transform
**Is your feature request related to a problem? Please describe.**
Need to add `PyTorch Tensor` support for `DataStats` transform to debug post transforms.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #681 .

### Description
This PR added PyTorch Tensor support for DataStats transform to help debug post transforms.

### Status
**Ready**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

File: monai/transforms/utility/array.py [hunk 1]
```diff
@@ -283,8 +285,13 @@ def __call__(
...
             lines.append(f"Shape: {img.shape}")
-        if self.intensity_range if intensity_range is None else intensity_range:
-            lines.append(f"Intensity range: ({np.min(img)}, {np.max(img)})")
+        if self.value_range if value_range is None else value_range:
+            if isinstance(img, np.ndarray):
+                lines.append(f"Value range: ({np.min(img)}, {np.max(img)})")
+            elif torch.is_tensor(img):
+                lines.append(f"Value range: ({torch.min(img)}, {torch.max(img)})")
+            else:
+                raise RuntimeError(f"unsupported input data type: {type(img)}.")
```
Comment (wyli) [inline monai/transforms/utility/array.py:294]: I don't think this should be a runtime error, could print "not a pytorch/numpy array"

Comment (Nic-Ma) [reply monai/transforms/utility/array.py]: updated in the latest commit.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for your review.
I updated the PR according to your comments.
Thanks.

**Pull request merged**